docs/source/sysadmin_guide/configure_sgx.rst

```diff
@@ -0,0 +1,458 @@
+
+.. code-block:: console
+
+    $ mkdir ~/sgx && cd ~/sgx
+    $ wget https://download.01.org/intel-sgx/linux-1.8/sgx_linux_x64_driver_4b57088.bin
+    $ chmod +x sgx_linux_x64_driver_4b57088.bin
+    $ sudo ./sgx_linux_x64_driver_4b57088.bin
+
+Download and install the Intel Capability Licensing Client. This is presently
+available only as an .rpm, so you must convert it to a .deb package with
+alien:
+
+.. code-block:: console
+
+    $ wget http://registrationcenter-download.intel.com/akdlm/irc_nas/11414/iclsClient-1.45.449.12-1.x86_64.rpm
+    $ sudo alien --scripts iclsClient-1.45.449.12-1.x86_64.rpm
+    $ sudo dpkg -i iclsclient_1.45.449.12-2_amd64.deb
+
+Download and install the Dynamic Application Loader Host Interface (JHI):
+
+.. code-block:: console
+
+    $ wget https://github.com/01org/dynamic-application-loader-host-interface/archive/master.zip -O jhi-master.zip
+    $ unzip jhi-master.zip && cd dynamic-application-loader-host-interface-master
+    $ cmake .
+    $ make
+    $ sudo make install
+    $ sudo systemctl enable jhi
+
+Download and install the Intel SGX Platform Software (PSW):
+
+.. code-block:: console
+
+    $ cd ~/sgx
+    $ wget https://download.01.org/intel-sgx/linux-1.8/sgx_linux_ubuntu16.04.1_x64_psw_1.8.100.37689.bin
+    $ chmod +x sgx_linux_ubuntu16.04.1_x64_psw_1.8.100.37689.bin
+    $ sudo ./sgx_linux_ubuntu16.04.1_x64_psw_1.8.100.37689.bin
+
+Check to make sure the kernel module is loaded:
+
+.. code-block:: console
+
+    $ lsmod | grep sgx
+    isgx                   36864  2
+
+If the output does not show the isgx module, make sure that
+SGX is set to "Enabled" in the BIOS.
+
+If you're still having trouble, the SGX software may need to be reinstalled:
+
+.. code-block:: console
+
+    $ sudo /opt/intel/sgxpsw/uninstall.sh
+    $ cd ~/sgx
+    $ sudo ./sgx_linux_x64_driver_4b57088.bin
+    $ sudo ./sgx_linux_ubuntu16.04.1_x64_psw_1.8.100.37689.bin
+
+After ensuring that the SGX kernel module is loaded, go to the next section
+to install and configure Sawtooth.
+
+
+Configuring Sawtooth to Use SGX
+===============================
+
+Install Sawtooth
+----------------
+
+.. code-block:: console
+
+    $ sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 8AA7AF1F1091A5FD
+    $ sudo add-apt-repository 'deb http://repo.sawtooth.me/ubuntu/0.8/stable xenial universe'
+    $ sudo apt-get update
+    $ sudo apt-get install -y -q \
+      sawtooth \
+      python3-sawtooth-poet-sgx
+
+
+Configure the validator to use SGX PoET
+---------------------------------------
+
+After installing Sawtooth, add config settings so PoET will work properly.
+Create the file ``/etc/sawtooth/poet_enclave_sgx.toml``
+with your favorite editor (such as vi):
+
+.. code-block:: console
+
+    $ sudo vi /etc/sawtooth/poet_enclave_sgx.toml
+
+Add the following lines:
+
+.. code-block:: console
+
+    # spid is a 32-digit hex string tied to the enclave implementation
+
+    spid = 'B785C58B77152CBE7FD55EE3851C4990'
```

Comment:
This is the Service Provider ID. It is linked to the key pair used to authenticate with the attestation service.